### PR TITLE
Add `golangci-lint`

### PR DESCRIPTION
**Pull request by joshuaspence**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    branches:
+      - "main"
+    tags:
+      - "v*"
+
+env:
+  GO_VERSION: "1.16"
+
+jobs:
+  build:
```
Comment (paultyng) [inline .github/workflows/ci.yaml:14]: Should we remove this one from acctest?

**Review by paultyng — APPROVED** [review]

File: .github/workflows/acctest.yml [hunk 2]
```diff
@@ -9,22 +9,6 @@ on:
...
-
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v2.3.4
-
-      - name: Build
-        run: make build
-
   test:
     name: Matrix Test
     needs: build
```
Comment (paultyng) [inline .github/workflows/acctest.yml:14]: Need to remove this as well I think.

**Review by paultyng — CHANGES_REQUESTED** [review]
Need to remove the dependency in acctest, see comment inline

**Comment by joshuaspence** [comment]
Ah, I saw that the checks all passed but didn't see that `acctest` was now invalid.

**Pull request merged**